src/App.tsx

```diff
@@ -1,24 +1,65 @@
 import React from 'react';
 import './App.scss';
+import Clock from './Components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  randomName: string;
+  isClockVisible: boolean;
 };
 
+type Props = {};
+
+// eslint-disable-next-line react/prefer-stateless-function
+class App extends React.Component<Props, State> {
+  state: State = {
+    randomName: 'randomNum',
+    isClockVisible: true,
+  };
+
+  componentDidUpdate(_prevProps: Props, prevState: State) {
+    if (prevState.randomName !== this.state.randomName) {
+      // eslint-disable-next-line no-console
+      console.log(`The Clock was renamed from ${prevState.randomName} to ${this.state.randomName}`);
+    }
+  }
+
+  render() {
+    const handleSwitcher = (typeBtn: string) => {
+      if (typeBtn === 'on') {
+        this.setState({ isClockVisible: true });
```

Comment:
```suggestion
        this.setState(prev => ({ isClockVisible: !prev.isClockVisible }));
```
Тебе не нужны будут if, если ты просто будешь менять текущее значение на противоположное